File: gilmorales/src/main/java/vistas/VistaPublicacionesGustadasUsaurioPublicoItem.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.html.Label;

/**
 * A Designer generated component for the vista-publicaciones-gustadas-usaurio-publico-item template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("vista-publicaciones-gustadas-usaurio-publico-item")
@JsModule("./src/vista-publicaciones-gustadas-usaurio-publico-item.ts")
public class VistaPublicacionesGustadasUsaurioPublicoItem extends LitTemplate {

    @Id("layoutItem")
	private Element layoutItem;
	@Id("publicacion")
	private Element publicacion;
	@Id("video")
	private Element video;
	@Id("numVisualizaciones")
	private Label numVisualizaciones;

	/**
     * Creates a new VistaPublicacionesGustadasUsaurioPublicoItem.
     */
    public VistaPublicacionesGustadasUsaurioPublicoItem() {
        // You can initialise any data required for the connected UI components here.
    }

}

File: gilmorales/frontend/src/vista-publicaciones-gustadas-usaurio-publico-item.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';

@customElement('vista-publicaciones-gustadas-usaurio-publico-item')
export class VistaPublicacionesGustadasUsaurioPublicoItem extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout style="align-items: center;" id="layoutItem">
 <vaadin-vertical-layout id="publicacion">
  <vaadin-vertical-layout theme="spacing" id="video"></vaadin-vertical-layout>
  <label id="numVisualizaciones">0</label>
 </vaadin-vertical-layout>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}
